packages/booter-lb3app/package.json

```diff
@@ -25,9 +25,9 @@
     "tslib": "^1.11.1"
   },
   "peerDependencies": {
-    "@loopback/boot": "^1.2.2",
-    "@loopback/core": "^1.5.0",
-    "@loopback/rest": "^1.10.3"
+    "@loopback/boot": "^1.2.2 || ^2.0.0",
+    "@loopback/core": "^1.5.0 || ^2.0.0",
+    "@loopback/rest": "^1.10.3 || ^2.0.0  || ^3.0.0"
```

Comment:
I'm also not very confident with the wide range as the CI only tests the module against latest versions of such dependencies.

The most reliable collection of `@loopback/*` modules that are guaranteed to work together are these being released together :-).

Reply:
@raymondfeng I think users will expect peer deps that meets the folowing criteria would work:

- Same major semver
- Equal or higher that min. version within semver (e.g. `@loopback/boot: ^1.2.2`)

If there's incompatibilities between 2 versions that meet the aforementioned criteria then we are breaking semver conventions.

So it may be a good idea to keep this range instead of pinning it to an exact version.

Reply:
@achrinza I meant to suggest we use `"@loopback/rest": "^3.0.0"`. It's hard to ensure that this module is compatible with 1.x, 2.x, and 3.x without meaningful CI test coverage.

Reply:
Take our connectors as an example. They support both LB3 and LB4, therefore we have only one version to maintain for both LB3 and LB4 users.

I think it makes a lot of sense to adopt the same approach for our extensions once we start maintaining LB4 LTS versions (see https://github.com/strongloop/loopback-next/issues/4398). This may include writing acceptance tests to verify that the current (`master`) version of an extension works with older (LTS) versions of a core component, as @raymondfeng has mentioned in his comment.

However, we don't provide LTS for LB4 yet, therefore I think it's ok to use a narrow range for the peer dependencies for now. Especially considering that we don't have CI test coverage to verify compatibility with older versions. I don't think it would be a good use of our time to work on such CI setup now.

If we decide to limit the supported peer versions to the latest major version only, then can we please find a way how to automate the update process - ideally as part of our version release setup? I.e. when we publish a new version, then the peer dependencies should be updated to match the new versions created by the publishing.